### PR TITLE
[python-nextgen] Add @overload for api methods to handle async requests

### PR DESCRIPTION
**Pull request by dcermak**
This PR adds overloads for the API call methods so that python tooling can properly infer whether you are calling an asynchronous method or not. The whole thing is unfortunately a little counter-intuitive, as `async_req=True` means that the request will not be using async-await, but will run in a threadpool.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc @spacether @krjakbrjak

### CODE REVIEW

**Comment by wing328** [comment]
@dcermak thanks for the PR.

cc @taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @tomplus (2018/10) @arun-nalla (2019/11) @spacether (2019/11) @krjakbrjak (2023/02)

File: samples/client/echo_api/python-nextgen/openapi_client/api/query_api.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 from pydantic import validate_arguments, ValidationError
 from typing_extensions import Annotated
 
+
```
Comment (tomplus) [inline samples/client/echo_api/python-nextgen/openapi_client/api/query_api.py:21]: I wonder if it's possible to remove this empty line. Linters may complain about it.

**Review by tomplus — APPROVED** [review]
lgtm

IMO we can go further and remove the thread pool from the async version. It'd be a braking change but using async_req in async methods are not a common pattern.

**Comment by dcermak** [comment]
> IMO we can go further and remove the thread pool from the async version. It'd be a braking change but using async_req in async methods are not a common pattern.

Imho it doesn't really make sense to use the thread pool in the async version as well. Additionally it is extremely counter-intuitive that `api_method(async_req=True)` does not return something that has to be `await`ed, but `api_method(async_req=False)` does…

I can try to remove it from the async version, but maybe in another PR? Or should this go in with this one as well?

**Comment by wing328** [comment]
> I can try to remove it from the async version, but maybe in another PR? Or should this go in with this one as well?

I suggest doing it in other PR.

Please update the samples and will merge this if all tests pass.

**Comment by dcermak** [comment]
> Please update the samples and will merge this if all tests pass.

Weird, I thought I updated all of them. Should be fixed now.

**Comment by dcermak** [comment]
The test failures have not been caused by my change, they are already present on the master branch. I have tried to find out how to fix them (it should be trivial, the typescript generator started generating `.gitignore` files, but I haven't found the right place…)

**Comment by wing328** [comment]
right, i'm looking into it. please ignore it for the time being.

**Comment by wing328** [comment]
build issues fixed in the master

**Pull request merged**